### PR TITLE
Add dedicated query for assessment form definition lookup

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Add a dedicated query for looking up the form definition to use for an assessment.

This is needed to support multiple custom assessments https://github.com/greenriver/hmis-frontend/pull/619

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/graphql/types/hmis_schema/query_type.rb [hunk 1]
```diff
@@ -171,18 +171,44 @@ def file(id:)
...
-      record.filter_context = { project: project }
+      record&.filter_context = { project: project } # Apply project-specific filtering rules. Only relevant for some form types.
+      record
+    end
+
+    field :assessment_form_definition, Types::Forms::FormDefinition, 'Get the correct Form Definition to use for an assessment, by Role or FormDefinition ID', null: true do
+      argument :project_id, ID, required: true, description: 'Project to use for Rule filtering on the definition'
+      argument :id, ID, required: false, description: 'Form Definition ID, if known'
+      argument :role, Types::Forms::Enums::AssessmentRole, required: false, description: 'Assessment role, if looking up by role'
+      argument :assessment_date, GraphQL::Types::ISO8601Date, required: false, description: 'Date to use for Rule filtering on the definition'
```
Comment (ttoomey) [inline drivers/hmis/app/graphql/types/hmis_schema/query_type.rb:191]: maybe validate that we only get one arg?
```suggestion
      argument :assessment_date, GraphQL::Types::ISO8601Date, required: false, description: 'Date to use for Rule filtering on the definition'
        validates required: { one_of: [:id, :role] }

```

File: drivers/hmis/app/graphql/types/hmis_schema/query_type.rb [hunk 2]
```diff
@@ -171,18 +171,44 @@ def file(id:)
       Hmis::File.viewable_by(current_user).find_by(id: id)
     end
 
-    field :get_form_definition, Types::Forms::FormDefinition, 'Get most relevant/recent form definition for the specified Role and project (optionally)', null: true do
+    field :get_form_definition, Types::Forms::FormDefinition, 'Get the most relevant Form Definition to use for record viewing/editing', null: true do
       argument :role, Types::Forms::Enums::FormRole, required: true
-      argument :enrollment_id, ID, required: false
       argument :project_id, ID, required: false
     end
+    def get_form_definition(role:, project_id: nil)
```
Comment (ttoomey) [inline drivers/hmis/app/graphql/types/hmis_schema/query_type.rb]: do you think it's worth it to rename this? Maybe `record_form_definition` ?

**Review by ttoomey — APPROVED** [review]
Looks good!

File: drivers/hmis/app/graphql/types/hmis_schema/query_type.rb (hunk 2 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/query_type.rb]: sure we might as well

**Comment by gigxz** [comment]
Addressed feedback with https://github.com/greenriver/hmis-warehouse/pull/3948/commits/f3de73998f265193fe538af4a80881c6d06757e4 + added separate Enum for the `recordFormDefinition` arg, plus descriptions for each role enum

**Pull request merged**